test/e2e/e2e_test.go

```diff
@@ -0,0 +1,119 @@
+	"github.com/loft-sh/vcluster/cmd/vclusterctl/flags"
+	"github.com/loft-sh/vcluster/cmd/vclusterctl/log"
+	logutil "github.com/loft-sh/vcluster/pkg/util/log"
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestRunE2ETests(t *testing.T) {
+	gomega.RegisterFailHandler(ginkgo.Fail)
+	ginkgo.RunSpecs(t, "e2e suite")
+}
+
+var _ = ginkgo.Describe("e2e test", func() {
+	ginkgo.Context("E2E", func() {
+
+		ginkgo.BeforeEach(func() {
+		})
+
+		ginkgo.It("Creates VirtualCluster successfully", func() {
+			ctx := context.TODO()
+			ctrl.SetLogger(logutil.NewLog(0))
+			l := log.GetInstance()
+			scheme := runtime.NewScheme()
+
+			// run port forwarder and retrieve kubeconfig for the vcluster
+			vKubeconfigFile, err := os.CreateTemp(os.TempDir(), "vcluster_e2e_kubeconfig_")
+			if err != nil {
+				errMsg := fmt.Errorf("could not create a temporary file: %w", err)
+				ginkgo.Fail(errMsg.Error())
+			}
+			// TODO: cleanup
+			// vKubeconfigFile removal is done in the Framework.Cleanup() which gets called in ginkgo's AfterSuite()
+
+			connectCmd := cmd.ConnectCmd{
+				Log: l,
+				GlobalFlags: &flags.GlobalFlags{
+					Namespace: "vcluster",
+					Debug:     true,
+				},
+				KubeConfig: vKubeconfigFile.Name(),
+				LocalPort:  14550, // choosing a port that usually should be unused
+			}
+			err = connectCmd.Connect(ctx, "vcluster", nil)
+			if err != nil {
+				l.Fatalf("failed to connect to the vcluster: %v", err)
+			}
+
+			var vclusterConfig *rest.Config
+			var vclusterClient *kubernetes.Clientset
+			var vclusterCRClient client.Client
+
+			err = wait.PollUntilContextTimeout(ctx, time.Second, time.Minute*5, false, func(ctx context.Context) (bool, error) {
+				output, err := os.ReadFile(vKubeconfigFile.Name())
+				if err != nil {
+					return false, nil
+				}
+
+				// try to parse config from file with retry because the file content might not be written
+				vclusterConfig, err = clientcmd.RESTConfigFromKubeConfig(output)
+				if err != nil {
+					return false, err
+				}
+				vclusterConfig.Timeout = time.Minute
+
+				// create kubernetes client using the config retry in case port forwarding is not ready yet
+				vclusterClient, err = kubernetes.NewForConfig(vclusterConfig)
+				if err != nil {
+					return false, err
+				}
+
+				vclusterCRClient, err = client.New(vclusterConfig, client.Options{Scheme: scheme})
+				if err != nil {
+					return false, err
+				}
+
+				// try to use the client with retry in case port forwarding is not ready yet
+				_, err = vclusterClient.CoreV1().ServiceAccounts("default").Get(ctx, "default", metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
```

Comment:
If you return an error here, the PollUntilContextTimeout function will imediately return with that error.
Is this intended or do you want to retry, until it works?

Reply:
it is ok to fail